### ISSUE TITLE
the Solar Frontier Inverter integration made for Home Assistant

### ISSUE THREAD

**Issue by djalexnl** [coen800/pysolarfrontier#2]
the Solar Frontier Inverter integration made for Home Assistant where can we find this?

Am a HomeAssistant User and also have this solar frontier converter. 
Running HASIO on a RapberryPi4 

I would like to use this Integration, but cannot find it anywhere?

Thanx in advanced Alex
(Netherlands)

**Comment by coen800** [comment]
Home assistant heeft mijn integration nooit toegevoegd daarom is hij niet te vinden. Je kan hem wel al custom component installeren, heb ik uiteindelijk ook gedaan.

**Comment by djalexnl** [comment]
Via Hacs bedoel je? als ik daar de url invoer krijg ik  volgende...

<Integration coen800/pysolarfrontier> Repository structure for v1.1 is not compliant

Kan je me helpen dit voor elkaar te krijgen, denk dat er meerdere zijn die daar mee geholpen zullen worden.
Dank voor je snelle reply etc. 
Gr Alexander

**Comment by coen800** [comment]
Nee HACS is iets anders. Je kan custom components toevoegen door deze te plaatsten in config/custom_components/
Ik heb een extra folder gemaakt in deze git met de bestanden die je nodig hebt.
Pad wordt config/custom_components/solar_frontier/FILES

Voeg dit toe aan je configuration.yaml:
```
sensor:
    # Solar frontier 
  - platform: solar_frontier
    host: 192.168.x.x
```

Reboot en als het goed is werkt het dan.

**Comment by djalexnl** [comment]
Dank voor je reponse... heb de 3 files vanuit de zip in  config/custom_components/solar_frontier geplaatst.
Daarna gereboot zodat ha de integration zou vinden... daarna inde configuration.yaml de host toegevoegdm met de juiste ip maar krijg volgende melding..

Logger: custom_components.solar_frontier.sensor
Source: custom_components/solar_frontier/sensor.py:50
Integration: solar_frontier
First occurred: 18:20:07 (1 occurrences)
Last logged: 18:20:07

Unexpected response received from Solar Frontier. Original error: Server disconnected

Zijn die 3 bestanden voldoende om de boel draaiende te krijgen? Het ipadres klopt als ik hem invul in de browser zie ik direct het display van mijn solarfrontier. Wellicht heb je een idee. Gr Alexander

**Comment by coen800** [comment]
Het zou kunnen dat de interface van je inverter anders is dan degene waar ik heb op getest heb, in mijn geval een SF-WR-3000. Zou je een screenshot kunnen maken van je interface? Dan kan ik zien of deze het zelfde is. Of heb je ook een SF-WR-3000?

**Comment by thomashogema** [comment]
Hi @coen800, 

Mooi om te zien dat je al een integratie hebt geschreven voor solar frontier! Ik weet niet zeker welk model omvormer wij hebben, komt de onderstaande overeen met de interface bij jou? 

![Screenshot_20221107-120550.png](https://user-images.githubusercontent.com/15984767/200319436-ba3ba8c0-268a-4b5b-81ed-6f78b1468bdb.png)

**Comment by coen800** [comment]
Nee, lijkt er wel op maar die van mij ziet er anders uit. Je zou het even moeten testen om zeker te weten of het werkt.

**Comment by thomashogema** [comment]
Hi @coen800,

Omdat Solar Frontier zijn online services gaat uitschakelen lijkt deze integratie een perfect alternatief. 

Helaas werkt hij voor mij wat inconsistent. Enkele dagen deed hij soms niks en de afgelopen twee weken is er geen enkele data binnengekomen. Er staat niks bijzonders in de logs, ook een herstart mag niet baten. Enig idee waar ik kan beginnen met troubleshooten?

![image](https://user-images.githubusercontent.com/15984767/227647390-b663b816-806c-4d19-b0fa-3b788ee9f2ec.png)

**Comment by thomashogema** [comment]
Na wat beter kijken viel op dat de sensoren (b.v. `solar_frontier_today_yield`) helemaal niet aangemaakt zijn. Er stond ook nog een `Setup of sensor platform solar_frontier is taking over 60 seconds` in de logs. 

Na 2x herstarten wordt de sensor wel netjes aangemaakt. Dit is alleen niet iedere herstart het geval 🤔.

**Comment by coen800** [comment]
Wat is het type omvormer wat je hebt?
De component maakt calls via HTTP naar bijvoorbeeld http://<IP>/gen.yield.day.chart.js en haalt op basis van de data in dit bestand de opbrengst op. Kan het wellicht zijn dat de omvormer op bepaalde momenten niet bereikbaar is via HTTP?

**Comment by thomashogema** [comment]
Dank voor de snelle reply!

Het gaat om de Turbo 1P PowerSet 4.1. Deze hangt gewoon bekabeld aan het netwerk en zou dus altijd beschikbaar moeten zijn. Tijdens een periode dat er geen statistieken zijn (b.v. hierboven op 19 februari) was de omvormer gewoon beschikbaar. Mogelijk dat dit tijdens het opstarten van de integratie net niet het geval was. 

Afgelopen 5 dagen liep de integratie probleemloos. Ik monitor de status nu wat beter, dus als hij weer wegvalt heb ik hopelijk meer informatie.

Jij hebt nooit vergelijkbare problemen gehad?
